src/components/CheckIn/AddPieceModal.js

```diff
@@ -47,20 +52,68 @@ const footer = (close, save, checkIn) => (
   </ModalFooter>
 );
 
+const FIELD_ATTRS_FOR_REQUIRED_LOCATION = {
+  required: true,
+  validate: Required,
+};
+
 class AddPieceModal extends Component {
   static propTypes = {
+    change: PropTypes.func.isRequired,
     checkIn: PropTypes.func.isRequired,
     close: PropTypes.func.isRequired,
+    createInventoryValues: PropTypes.object.isRequired,
+    dispatch: PropTypes.func.isRequired,
     handleSubmit: PropTypes.func.isRequired,
     locations: PropTypes.arrayOf(PropTypes.shape({
       label: PropTypes.string.isRequired,
       value: PropTypes.string.isRequired,
     })),
     showPieceFormatField: PropTypes.bool,
     store: PropTypes.object.isRequired,
+    submitting: PropTypes.func.isRequired,
+  }
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      isAddItemButtonDisabled: true,
+      validateLocation: {},
+    };
   }
 
-  checkIn = () => this.props.checkIn(getFormValues('AddPieceModalForm')(this.props.store.getState()));
+  componentDidMount() {
+    this.validateRequiredFields();
+  }
+
+  checkIn = () => this.props.checkIn(getFormValues(ADD_PIECE_MODAL_FORM)(this.props.store.getState()));
+
+  onChangeSelect = (_, value, prevValue, fieldName) => {
+    const { dispatch, change } = this.props;
+
+    dispatch(change(fieldName, value));
+    this.validateRequiredFields();
+  }
+
+  validateRequiredFields = () => {
+    const { store, createInventoryValues } = this.props;
+    const formValues = getFormValues(ADD_PIECE_MODAL_FORM)(store.getState());
+
+    if (includes(createInventoryValues[formValues.format], INVENTORY_RECORDS_TYPE.all ||
+      INVENTORY_RECORDS_TYPE.instanceAndHolding)) {
+      this.setState({ validateLocation: FIELD_ATTRS_FOR_REQUIRED_LOCATION });
+    } else {
+      this.setState({ validateLocation: {} });
+    }
+
+    if (formValues.locationId &&
+      includes(createInventoryValues[formValues.format], INVENTORY_RECORDS_TYPE.instanceAndHolding ||
+      INVENTORY_RECORDS_TYPE.all)) {
+      this.setState({ isAddItemButtonDisabled: false });
+    } else {
+      this.setState({ isAddItemButtonDisabled: true });
+    }
```

Comment:
It might be better to combine both setState calls in one place?

Reply:
yep, it should be one `setState`, but discussed this function offline

Reply:
You're right. Updated

Reply:
hm, should it be this.setState({ isAddItemButtonDisabled: expression from if`?

Reply:
Removed state at all